src/dev/riscv/HiFive.py

```diff
@@ -111,6 +112,14 @@ def _off_chip_devices(self):
         """Returns a list of off-chip peripherals"""
         return []
 
+    def _on_chip_ports(self):
+        """Returns a list of on-chip peripherals request/response ports"""
+        return []
+
+    def _off_chip_ports(self):
+        """Returns a list of off-chip peripherals request/response ports"""
+        return []
+
```

Comment:
Not a hard requirement for this PR as we didn't have these in the HiFive code prior to this PR, but it would be nice if the functions included typing hints for return types.

e.g.:

```suggestion
    def _on_chip_ports(self) -> List[Port]:
        """Returns a list of on-chip peripherals request/response ports"""
        return []

    def _off_chip_ports(self) -> List[Port]:
        """Returns a list of off-chip peripherals request/response ports"""
        return []

```

Reply:
[Added](https://github.com/gem5/gem5/pull/2087/files#diff-ac1ee32f2b73ad1a3c9faadab59817e09670fb22a3ae2a080bfba892035a6a3f)